backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/BaseController.java

```diff
@@ -46,6 +46,9 @@ protected Transaction transactionStart(HttpServletRequest request, TransactionTy
 	 * @param identifier The value of the identifier
 	 */
 	protected void addAffectedParty(Transaction transaction, IdentifierType type, String identifier, AffectedPartyDirection direction) {
+		if (identifier == null || identifier.isBlank() || type == null || direction == null) {
```

Comment:
You can use StringUtils.isEmpty() in place of the first two checks.

The type and direction checks are probably not required as the only reason this would happen is through bad coding. If you want to validate a method against bad usage you would typically throw an IllegalArgumentException though I don't think it's necessary in this case as we aren't using them consistently in the code base. Which isn't a horrible thing as this isn't library/api code.

If the identifier is null it also might be useful to add a logger.info message. That way there's a record of the fact that the auditing couldn't be done.